spec/unit/we/call/connection_spec.rb

```diff
@@ -131,7 +137,25 @@
         end
 
         it 'is not repeated adapter handler' do
-          expect(subject.builder.handlers.map(&:klass)).to contain_exactly(described_class::DEFAULT_ADAPTER_CLASS, Faraday::Sunset)
+          expect(handlers).to match_array(
+            [described_class::DEFAULT_ADAPTER_CLASS].concat(DEFAULT_MIDDLEWARES)
+          )
+        end
+      end
+
+      context 'when :net_http adapter is specified' do
```

Comment:
I want to suggest that your tests for this would not be adapter oriented, but aspect oriented.
So i would have something like
```
context 'when adapter supports gzip' do
   let(:gzip_supported_adapter) { Faraday::Adapter::NetHttp } #or move the array you have in code into a const and use it here to 'sample' one of those gzip supporting adapters
   it 'should not include the gzip handler' do
      ....
   end
end

context 'when adapter does not support gzip` do
   ... #same concept just the opposite
end
```

It doesn't really matter which one is the default adapter, that might change any day and you don't want you tests to break because of that

Reply:
It very much matter which one is the default, we’ve seen issues before when the unexpected adapter was in place, and that’s an awful position to be in. We want to confirm that when nothing is set, typheous specifically is enabled. 

If the default changes that will be a big deal, and I won’t mind taking a minute or two to change the test.

Other than that, I’m happy to change things to “aspect based”.

Reply:
I didn't really mean that it doesn't matter what the default is, i meant that from a tests standpoint, the tests that you wrote for the default adapter are actually tests for Typhoeus.

Having a set of tests for Typhoeus + a test which verifies that when no other adapter is explicitly set, the used adapter is Typhoeus, would make more sense to me

Reply:
I didn't write any tests for the default adapter, I didn't write any tests for typheous. This test is confirming that the default adapter has not been registered twice, which is a regression against an actual bug we had for end users specifying adapters themselves. The registered middleware was `[Faraday::Adapter::NetHttp, Faraday::Adapter::NetHttp, Faraday::Sunset]`, which meant it made all HTTP requests twice. 

It doesn't do that now, and this test is there to make sure of it.

Reply:
Got it. Sorry for the false accusations, i must have been out of focus or something 😄